js/dataprocesser_worker.js

```diff
@@ -202,7 +202,26 @@ LayerWorker.TexBuilder = {
                                                 ptexture.glcontext,
                                                 false,
                                                 ptexture.isMask);
-          return node;
+          if (ptexture.isMask) {
```

Comment:
Pass ptexture into LayerScope.TextureNode constructor function.
And do all the things bellow there